src/Sylius/Component/Core/Provider/TranslationLocaleProvider.php

```diff
@@ -29,12 +29,20 @@ public function getDefinedLocalesCodes(): array
     {
         $locales = $this->localeRepository->getAll();
 
-        return array_map(
-            function (LocaleInterface $locale) {
+        $codes = array_map(
+            static function (LocaleInterface $locale): string {
                 return (string) $locale->getCode();
             },
             $locales,
         );
+
+        $default = $this->defaultLocaleCode;
+        if (false !== ($key = array_search($default, $codes, true))) {
+            unset($codes[$key]);
+            array_unshift($codes, $default);
+        }
+
+        return $codes;
```

Comment:
There's no need to process the array if the default locale is already positioned first.
```suggestion
        $key = array_search($default, $localeCodes, true);

        
        if ($key === false || $key === 0) {
            return $localeCodes;
        }

        unset($localeCodes[$key]);
        array_unshift($localeCodes, $default);

        return $localeCodes;

```

Reply:
👍🏻  Fixed